### PR TITLE
Support date_add/date_sub expression functions

### PR DESCRIPTION
**Pull request by gengliqi**
### What is changed and how it works?

Issue Number: Close https://github.com/tikv/tikv/issues/17529

What's Changed:

```commit-message
Support date_add/date_sub expression functions
```

### Related changes

- [x] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
Support date_add/date_sub expression functions
```

### CODE REVIEW

File: components/tidb_query_datatype/src/codec/mysql/time/interval.rs [hunk 1]
```diff
@@ -0,0 +1,1774 @@
...
+            ("2", Month),
+            ("-2", Month),
+            ("99", Quarter),
+            ("-99", Quarter),
+            ("99", Year),
+            ("-99", Year),
+            ("-34 23:59:59.1234", DayMicrosecond),
+        ];
+        for (input, unit) in err_cases {
+            let result = Interval::extract_duration(&mut ctx, &unit, input);
```
Comment (overvenus) [inline components/tidb_query_datatype/src/codec/mysql/time/interval.rs:1770]: You can use proptest[1] for testing parsing methods, 

\[1\]: https://proptest-rs.github.io/proptest/proptest/getting-started.html

**Review by overvenus — APPROVED** [review]

**Review by breezewish — APPROVED** [review]

File: components/tidb_query_datatype/src/codec/mysql/time/interval.rs (hunk 1 above)
Comment (gengliqi) [reply components/tidb_query_datatype/src/codec/mysql/time/interval.rs:1770]: I will try it next time due to lack of time.

**Pull request merged**